holoviews/plotting/bokeh/plot.py

```diff
@@ -393,7 +393,11 @@ def sync_sources(self):
                 shared_sources.append(new_source)
                 source_cols[id(new_source)] = [c for c in new_source.data]
         for plot in plots:
-            for hook in plot.finalize_hooks:
+            if plot.hooks and plot.finalize_hooks:
+                self.warning("Supply either hooks or finalize_hooks not both, "
```

Comment:
```suggestion
                self.warning("Supply either hooks or finalize_hooks not both; "
```

Reply:
Are you sure you want to warn for this immediately, before e.g. pyviz.org is updated not to use finalize_hooks?  Maybe it doesn't use that, though.

Reply:
This is only about supplying both hooks and finalize_hooks, which no example in pyviz.org would do.